sos/plugins/grub2.py

```diff
@@ -33,10 +33,13 @@ def setup(self):
             "/etc/grub2.cfg",
             "/etc/grub.d"
         ])
-        self.add_cmd_output([
-            "ls -lanR /boot",
-            "grub2-mkconfig"
-        ])
+        self.add_cmd_output("ls -lanR /boot")
+        # grub2-mkconfig command will load btrfs and ext4 kernel modules
+        # if those modules are not loaded at the time of the command running
+        # This behaviour causes an unexpected configuration change for system.
+        # sosreport should aovid such situation.
+        if self.is_module_loaded("btrfs") and self.is_module_loaded("ext4"):
+            self.add_cmd_output("grub2-mkconfig")
```

Comment:
I don't really like this as a solution; there will be many systems on RHEL7 that do not meet the btrfs|ext4 conditions and where we will lose diagnostic information as a result of this change.

I think we need to take this back to `grub2` and `os-prober`.